pkg/asset/agent/image/ignition.go

```diff
@@ -387,3 +380,22 @@ func RetrieveRendezvousIP(agentConfig *agent.Config, nmStateConfigs []*v1beta1.N
 	}
 	return rendezvousIP, err
 }
+
+func getMirrorFromRelease(releaseImage string, registriesConfig *mirror.RegistriesConf) string {
+
+	releaseImageMirror := ""
+	source := regexp.MustCompile(`^(.+?)(@sha256)?:(.+)`).FindStringSubmatch(releaseImage)
+	for _, config := range registriesConfig.MirrorConfig {
+		if config.Location == source[1] {
+			// include the tag with the build release image
+			if len(source) == 4 {
+				// Has Sha256
+				releaseImageMirror = fmt.Sprintf("%s%s:%s", config.Mirror, source[2], source[3])
+			} else if len(source) == 3 {
+				releaseImageMirror = fmt.Sprintf("%s:%s", config.Mirror, source[2])
+			}
+		}
```

Comment:
Pre-existing issue, but it's a bit surprising that we don't return or break here, so we're always choosing the _last_ matching mirror on the list.